bcda/models/service.go

```diff
@@ -153,6 +156,27 @@ func (s *service) GetJobAndKeys(ctx context.Context, jobID uint) (*Job, []*JobKe
 	return j, nonEmptyKeys, nil
 }
 
+func (s *service) CancelJob(ctx context.Context, jobID uint) (uint, error) {
+	// Assumes the job exists and retrieves the job by ID
+	job, err := s.repository.GetJobByID(ctx, jobID)
+	if err != nil {
+		return 0, err
+	}
+
+	// Check if the job is pending or in progress.
+	if job.Status == JobStatusPending || job.Status == JobStatusInProgress {
+		job.Status = JobStatusCancelled
+		err = s.repository.UpdateJob(ctx, *job)
+		if err != nil {
```

Comment:
Nitpick but you could cut down on the returns this way by checking for err being nil
```
...
    if err == nil {
      return jobID, nil
    }
  }
  return 0, ErrJobNotCancelled
}
```

Reply:
This was from my code that Katie based her branch on, but this is a trick on the eyes.  There are 3 different returns here, the difference is`ErrJobNotCancelled` and `ErrJobNotCancellable`  This is not the first time getting them confused though.  Should I have a better name for one to better distinguish between them when reading through the code?

Reply:
I shouldn't look at PRs too late >_<